yapapi/executor/agreements_pool.py

```diff
@@ -202,3 +206,7 @@ async def on_agreement_terminated(self, agr_id: str, reason: dict) -> None:
             buffered_agreement.worker_task and buffered_agreement.worker_task.cancel()
             del self._agreements[agr_id]
             self.emitter(events.AgreementTerminated(agr_id=agr_id, reason=reason))
+
+    def last_agreement_rejected(self, provider_id: str) -> bool:
```

Comment:
You can change `class AgreementsPool:` to `AgreementsPool(ComputationHistory)`, so that it explicitly implements this protocol.